[2025-04-09T20:45:25.055+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-04-09T20:45:24.166399+00:00 [queued]>
[2025-04-09T20:45:25.059+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-04-09T20:45:24.166399+00:00 [queued]>
[2025-04-09T20:45:25.059+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-04-09T20:45:25.068+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): reddit_extraction> on 2025-04-09 20:45:24.166399+00:00
[2025-04-09T20:45:25.070+0000] {standard_task_runner.py:57} INFO - Started process 53 to run task
[2025-04-09T20:45:25.071+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'etl_reddit_pipeline', 'reddit_extraction', 'manual__2025-04-09T20:45:24.166399+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/reddit_dag.py', '--cfg-path', '/tmp/tmp0kazbj6v']
[2025-04-09T20:45:25.071+0000] {standard_task_runner.py:85} INFO - Job 4: Subtask reddit_extraction
[2025-04-09T20:45:25.091+0000] {task_command.py:416} INFO - Running <TaskInstance: etl_reddit_pipeline.reddit_extraction manual__2025-04-09T20:45:24.166399+00:00 [running]> on host 7645e7e94702
[2025-04-09T20:45:25.119+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Hari Shankar' AIRFLOW_CTX_DAG_ID='etl_reddit_pipeline' AIRFLOW_CTX_TASK_ID='reddit_extraction' AIRFLOW_CTX_EXECUTION_DATE='2025-04-09T20:45:24.166399+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-09T20:45:24.166399+00:00'
[2025-04-09T20:45:25.121+0000] {logging_mixin.py:151} WARNING - Version 7.7.1 of praw is outdated. Version 7.8.1 was released Friday October 25, 2024.
[2025-04-09T20:45:25.122+0000] {logging_mixin.py:151} INFO - connected to reddit!
[2025-04-09T20:45:25.304+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/pipelines/reddit_pipeline.py", line 9, in reddit_pipeline
    posts = extract_posts(instance, subreddit, time_filter, limit)
  File "/opt/airflow/etls/reddit_etl.py", line 23, in extract_posts
    for post in posts:
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/models/listing/generator.py", line 63, in __next__
    self._next_batch()
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/models/listing/generator.py", line 89, in _next_batch
    self._listing = self._reddit.get(self.url, params=self.params)
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/util/deprecate_args.py", line 43, in wrapped
    return func(**dict(zip(_old_args, args)), **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/reddit.py", line 712, in get
    return self._objectify_request(method="GET", params=params, path=path)
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/reddit.py", line 517, in _objectify_request
    self.request(
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/util/deprecate_args.py", line 43, in wrapped
    return func(**dict(zip(_old_args, args)), **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/praw/reddit.py", line 941, in request
    return self._core.request(
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/sessions.py", line 328, in request
    return self._request_with_retries(
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/sessions.py", line 234, in _request_with_retries
    response, saved_exception = self._make_request(
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/sessions.py", line 186, in _make_request
    response = self._rate_limiter.call(
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/rate_limit.py", line 46, in call
    kwargs["headers"] = set_header_callback()
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/sessions.py", line 282, in _set_header_callback
    self._authorizer.refresh()
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/auth.py", line 378, in refresh
    self._request_token(grant_type="client_credentials", **additional_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/auth.py", line 155, in _request_token
    response = self._authenticator._post(url=url, **data)
  File "/home/airflow/.local/lib/python3.9/site-packages/prawcore/auth.py", line 59, in _post
    raise ResponseException(response)
prawcore.exceptions.ResponseException: received 401 HTTP response
[2025-04-09T20:45:25.317+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=etl_reddit_pipeline, task_id=reddit_extraction, execution_date=20250409T204524, start_date=20250409T204525, end_date=20250409T204525
[2025-04-09T20:45:25.324+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 4 for task reddit_extraction (received 401 HTTP response; 53)
[2025-04-09T20:45:25.345+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-04-09T20:45:25.353+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
